metadata/db.go

```diff
@@ -315,11 +319,15 @@ func (s GCStats) Elapsed() time.Duration {
 func (m *DB) GarbageCollect(ctx context.Context) (gc.Stats, error) {
 	m.wlock.Lock()
 	t1 := time.Now()
+	ctx, span := tracing.StartSpan(ctx, "GarbageCollect")
+	defer tracing.StopSpan(span)
+
 	c := startGCContext(ctx, m.collectors)
 
 	marked, err := m.getMarked(ctx, c) // Pass in gc context
 	if err != nil {
 		m.wlock.Unlock()
+		span.RecordError(err)
```

Comment:
You are already recording errors returned from this method in scheduler.go in L329. We can perhaps rely on the top level RecordError and skip recording error in the metadata package.

Reply:
I think your suggestion is good.